web/src/designer/components/condition/ConditionModal.tsx

```diff
@@ -12,33 +12,88 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-import {Button, Dialog} from '@mui/material';
-import {useState} from 'react';
+import {
+  Button,
+  Dialog,
+  DialogActions,
+  DialogContent,
+  Stack,
+} from '@mui/material';
+import {useCallback, useEffect, useState} from 'react';
 import {ConditionControl} from './ConditionControl';
-import {ConditionProps} from './types';
+import {ConditionProps, ConditionType} from './types';
 import QuizIcon from '@mui/icons-material/Quiz';
 
 export const ConditionModal = (props: ConditionProps & {label: string}) => {
   const [open, setOpen] = useState(false);
+  // Local draft copy
+  const [draft, setDraft] = useState<ConditionType | null>(
+    props.initial ?? null
+  );
+
+  // Reset the draft whenever the parent value changes while closed
+  // Doing it this way because the modal exists even when closed
+  useEffect(() => {
+    if (!open) setDraft(props.initial ?? null);
+  }, [props.initial, open]);
+
+  // Open the dialog and refresh draft from parent prop
+  const handleOpen = () => {
+    setDraft(props.initial ?? null);
+    setOpen(true);
+  };
+
+  const close = () => setOpen(false);
+
+  // Warn when cancelling
+  const handleCancel = useCallback(() => {
+    const changed =
+      JSON.stringify(draft) !== JSON.stringify(props.initial || null);
+    if (changed) {
+      const confirm = window.confirm(
```

Comment:
Good idea but can you use a dialog component rather than the system confirm.  What's here is confusing since there is a 'cancel' button in the confirm dialog which cancels the cancellation...